requirements/customer/cli.tex

```diff
@@ -0,0 +1,102 @@
+However, some commands and options are shared between components or commands.
+
+The JobAdder CLI ignores capitalization.
+JobAdder CLI options override the values of configuration files.
+If no command is given, the general help text for the given component is printed out.
+\subsubsection{Shared Options}
+\jaoption{config}{c}{Overrides the global configuration file with the given configuration file.}
+\jaoption{detach}{d}{Detaches the execution of the given command from the command line.}
+\jaoption{help}{h}{
+Prints out a list of the available commands. 
+When used in conjunction with a command:
+prints out more detailed information on the given command and a list of the available options.}
+\jaoption{verbose}{v}{
+Determines the amount of information printed out.
+If set to 2, detailed information is printed out.
+If set to 1, only high-level information is printed out (default).
+If set to 0, no information is printed out.}
+\subsubsection{User Client}
+\jacommand{add}
+{gpu, mount, name, priority, server, ram, threads}
+{Sends one or more job requests to the server.
+If targets are Docker containers the worker client simply runs the container.
+If targets are configuration files with a specified terminal command the worker executes the given terminal command.
+Otherwise targets are interpreted as terminal commands}
+\jaoption{gpu}{g}{
+If set to true, informs the server that the job requires a gpu.
+If set to a string that can be interpreted as VRAM amount, informs the server that a GPU with this amount of VRAM is required for the job.
+If set to a specific GPU model, informs the server that the job must be run on a machine with that GPU built in.
+}
+\jaoption{mount}{m}
+{Directories that need to be mounted onto the Docker container that the job is running in.}
+\jaoption{name}{n}
+{Human-readable name of the job.
+Defaults to user's name + job index}
+\jaoption{priority}{p}
+{Sets the priority of the job: Very high, high, medium, or low.
+Defaults to medium.}
+\jaoption{server}{s}
+{Server to send the job request to.
+Defaults to localhost.}
+\jaoption{ram}{r}
+{If set to a string that can be interpreted as RAM amount, informs the server that this amount of RAM is required for the job.}
+\jaoption{threads}{t}
+{If set to an integer, informs the server that this number of threads is required for the job.}
+
+\jacommand{pause}{None}{Pauses the job with the specified name.}
+
+\jacommand{query}{user}
+{Prints out information on running/queued/past jobs (specified by target).
+In addition to the one below, options from add command can be used as constraints.}
+\jaoption{after}{a}{Only return jobs added after the given point in time.}
+\jaoption{before}{b}{Only return jobs added before the given point in time.}
+\jaoption{user}{u}{Filter query by user that added the job.}
+
+\jacommand{resume}{None}{Resumes the job with the specified name.}
```

Comment:
In which case does the client want to pause their own task?

Reply:
IIRC, we decided that a pause and resume option is not necessary. If a user wants to release resources to another, they could lower the priority of their job instead of pausing it.